test/unittests/test_ecc_functionality.cpp

```diff
@@ -0,0 +1,475 @@
+                    }
+                } else {
+                    if (op->getType() == qc::ClassicControlled) {
+                        auto*              cc_op          = dynamic_cast<qc::ClassicControlledOperation*>(op.get());
+                        const auto         start_index    = static_cast<unsigned short>(cc_op->getParameter().at(0));
+                        const auto         length         = static_cast<unsigned short>(cc_op->getParameter().at(1));
+                        const unsigned int expected_value = cc_op->getExpectedValue();
+                        unsigned int       actual_value   = 0;
+                        for (unsigned int i = 0; i < length; i++) {
+                            actual_value |= (classicValuesECC[start_index + i] ? 1u : 0u) << i;
+                        }
+                        if (actual_value != expected_value) {
+                            continue;
+                        }
+                    }
+                    auto operation = dd::getDD(op.get(), dd);
+                    rootEdge       = dd->multiply(operation, rootEdge);
+                }
+                if (simulateNoise && operationCounter == applyNoiseAfterNOperations) {
+                    dd->incRef(rootEdge);
+                    auto result = dd->measureOneCollapsing(rootEdge, targetForNoise, false, mt);
+                    if (result == '1') {
+                        auto flipOperation = StandardOperation(qc.getNqubits(), targetForNoise, qc::X);
+                        auto operation     = dd::getDD(&flipOperation, dd);
+                        rootEdge           = dd->multiply(operation, rootEdge);
+                    }
+                }
+                operationCounter++;
+            }
+            for (std::size_t i = 0; i < classicValuesECC.size(); i++) {
+                // Counting the final hit in resultReg after each shot
+                auto regName = qc.returnClassicalRegisterName(i);
+                if (regName == "resultReg") {
+                    if (finalClassicValues.count(i) == 0) {
+                        finalClassicValues[i] = 0;
+                    }
+                    if (classicValuesECC[i]) {
+                        finalClassicValues[i] += 1.0 / nrShots;
+                    }
+                }
+            }
+        }
+    }
+
+    static bool verifyExecution(qc::QuantumComputation& qcOriginal, qc::QuantumComputation& qcECC,
+                                bool                          simulateWithErrors     = false,
+                                const std::vector<dd::Qubit>& dataQubits             = {},
+                                int                           insertErrorAfterNGates = 0) {
+        double                        tolerance = 0.15;
+        double                        aboutOne  = 1.00000001;
+        std::map<std::size_t, double> finalClassicValuesOriginal{};
+        bool                          testingSuccessful = true;
```

Comment:
this variable should not be necessary. if one of the tests fails, then you can immediately return false. Doesn't make sense (especially, in the CI environment) to continue the testing.

Reply:
I changed the code as suggested.